### ISSUE TITLE
Setting `Display::None` doesn't hide UI elements

### ISSUE THREAD

**Issue by pascualex** [bevyengine/bevy#7952]
## Bevy version

`0.10`

## What you did

I updated an old project where I toggle the display of certain UI elements.

## What went wrong

Once they were set to `Display::Flex` for the first time, setting `Display::None` didn't hide them.

## Additional information

I built a minimal repro. The node spawns with `Display::None`. Pressing space for the first time toggles the display, showing the node. Pressing space a second time toggles the display but doesn't hide the node. This example works as expected when ran with Bevy `0.9`.

```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(spawn_camera_and_node)
        .add_system(toggle_display)
        .run();
}

fn spawn_camera_and_node(mut commands: Commands) {
    commands.spawn(Camera2dBundle::default());
    commands.spawn(NodeBundle {
        style: Style {
            display: Display::None,
            size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
            ..default()
        },
        background_color: Color::RED.into(),
        ..default()
    });
}

fn toggle_display(mut query: Query<&mut Style>, input: Res<Input<KeyCode>>) {
    if input.just_pressed(KeyCode::Space) {
        let mut style = query.single_mut();
        style.display = match style.display {
            Display::Flex => Display::None,
            Display::None => Display::Flex,
        };
    }
}
```

**Comment by rparrett** [comment]
Bisected to 76058bcf3

**Comment by nicoburns** [comment]
This one I can reproduce in Taffy. "leaf" display mode for nodes with no children is taking precedence over `display: none`. Fix incoming. Just trying to work out why our tests didn't catch this as we do have tests specifically testing `display: none`.

**Comment by nicoburns** [comment]
Ah. So this bug only triggers if the `display: none` node is the only node in the hierarchy. If it either has children OR is itself a child of another node then it work fine!

**Comment by nicoburns** [comment]
@pascualex Fix submitted in the form of #7953. In the meantime I believe you ought to be able to work around this issue by introducing an extra (useless) UI node.

**Comment by kelden** [comment]
@nicoburns I have an empty menu node that I have created in a startup system with Display::None. It is correctly hidden at first, but when I toggle it to Flex and back to None it stays visible. This happens with or without children nodes, after updating to Taffy 0.3.4 with `cargo update`.

**Comment by pascualex** [comment]
@nicoburns that was quick! Thank you.

**Comment by pascualex** [comment]
Yeah, @kelden is right, the minimal example is still failing for me too. Even though as you pointed `cargo update` does update Taffy to `0.3.4`.

**Comment by nicoburns** [comment]
This needs to be reopened, as it's not completely fixed.

**Comment by nicoburns** [comment]
@pascualex @kelden Taffy `v0.3.5` has now been release which I have confirmed fixes the minimal example for me. You should be able to get it by running `cargo update` again.

**Comment by pascualex** [comment]
@nicoburns yes, it works as expected now. Thanks again!
